devicetypes/fibargroup/fibaro-dimmer-2-zw5.src/fibaro-dimmer-2-zw5.groovy

```diff
@@ -170,6 +170,10 @@ def syncStart() {
 				state."$it.key".state = "notSynced"
 				syncNeeded = true
 			}
+			// this parameter (38) is not supported on some earlier firmware versions, so we'll mark it as already synced
+			if ("$it.key" == "levelCorrection" && (zwaveInfo.ver as float) <= REDUCED_CONFIGURATION_VERSION) {

```

Comment:
This could result in `syncNeeded` being incorrectly set to true if `levelCorrection` was the only remaining parameter that required sync. I suggest doing this check up above so `syncNeeded` is only set to `true` if the parameter needs to be synced.

Reply:
This still seems odd. Why not just move this up into the loop above and only set `syncNeeded` if sync is needed?

Reply:
I guess I'm misunderstanding what you're asking? There's no loop above this. Did you mean the if statement above? If so, this seems like the more prudent place to put it since it only adds one extra comparison.

Reply:
Sorry, I was reading the code wrong and making incorrect comments. What I suggest is moving this new `if` statement to above the `if (state."$it.key".value != settings."$it.key" as Integer || state."$it.key".state in ["notSynced","inProgress"])` and then turn that into an `else if`.

Reply:
Gotcha. Fixed now. And Fibaro got back to us with the correct fw version number.

Reply:
Changes look good. Is there any risk that `zwaveInfo.ver` won't be set? Just wondering if we should handle that being null and assume the FW version does not support this parameter.

Reply:
Don't expect it will ever be the case, but it can't hurt. Null check added.